kyo-core/shared/src/main/scala/kyo/Async.scala

```diff
@@ -299,6 +308,24 @@ object Async:
                     p
                 }
 
+            /** Maps the Result of the Fiber using the provided function.
+              *
+              * This method allows you to transform both the error and success types of the Fiber's result. It's useful when you need to
+              * modify the error type or perform a more complex transformation on the success value that may also produce a new error type.
+              *
+              * @param f
+              *   The function to apply to the Fiber's Result. It should take a Result[E, A] and return a Result[E2, B].
+              * @return
+              *   A new Fiber with the mapped Result
+              */
+            def mapResult[E2, B](f: Result[E, A] => Result[E2, B])(using Frame): Fiber[E2, B] < IO =
```

Comment:
Another option is `sandbox` function (`Fiber[E, A] => Fiber[Nothing, Result[E, A]`) + `flatten` which could flatten one of Promise/Fiber/Result?

Otherwise seems okay to me.

Reply:
I think the complexity of the API might not be worth it